joern-cli/frontends/rubysrc2cpg/src/main/scala/io/joern/rubysrc2cpg/astcreation/AstForTypesCreator.scala

```diff
@@ -60,16 +61,66 @@ trait AstForTypesCreator { this: AstCreator =>
     astExprOfCommand ++ Seq(bodyBlockAst)
   }
 
-  def astForClassOrModuleReferenceContext(
-    ctx: ClassOrModuleReferenceContext,
-    baseClassName: Option[String] = None
-  ): Seq[Ast] = {
-    val className = ctx.className(baseClassName)
+  def astForModuleDefinitionPrimaryContext(ctx: ModuleDefinitionPrimaryContext): Seq[Ast] = {
+    val className = ctx.moduleDefinition().classOrModuleReference().className(None)
 
     if (className != Defines.Any) {
       classStack.push(className)
+
+      val fullName = classStack.reverse.mkString(pathSep)
+      val namespaceBlock = NewNamespaceBlock()
+        .name(className)
+        .fullName(fullName)
+        .filename(filename)
+
+      val moduleBodyAst = astInFakeMethod(className, fullName, filename, ctx)
+
+      if (classStack.size > 0) {
+        classStack.pop()
+      }
+      Seq(Ast(namespaceBlock).withChildren(moduleBodyAst))
+    } else {
+      Seq.empty
     }
-    Seq(Ast())
+
+  }
+
+  private def astInFakeMethod(
+    name: String,
+    fullName: String,
+    path: String,
+    ctx: ModuleDefinitionPrimaryContext
+  ): Seq[Ast] = {
+
+    val fakeGlobalTypeDecl = NewTypeDecl()
+      .name(name)
+      .fullName(fullName)
+    val fakeGlobalMethod =
+      methodNode(ctx, name, name, fullName, None, path, Option(NodeTypes.TYPE_DECL), Option(fullName))
+    scope.pushNewScope(fakeGlobalMethod)
+    val blockNode_ = blockNode(ctx)
+
+    val methodReturn = methodReturnNode(ctx, Defines.Any)
+    val bodyStmtAsts = astForBodyStatementContext(ctx.moduleDefinition().bodyStatement())
+    val bodyAstSansModifiers = bodyStmtAsts
+      .filterNot(ast => {
+        val nodes = ast.nodes
+          .filter(_.isInstanceOf[NewIdentifier])
+
+        if (nodes.size == 1) {
+          val varName = nodes
+            .map(_.asInstanceOf[NewIdentifier].name)
+            .head
+          varName == "public" || varName == "protected" || varName == "private"
```

Comment:
Could probably look at trying something like `io.shiftleft.codepropertygraph.generated.ModifierTypes.ALL.contains(varName.toUpperCase)`

Reply:
It was already present, I just refactored it a big. 

I guess I need a respective test case to verify this one. I just updated the existing code that I moved from `AstCreator` to  `AstForTypesCreator`. Let me check with @rahul-privado

Reply:
I would need respective unit tests to test this change. I will handle this in a separate PR.